tests/impute/test_fill_forward_work_columns.py

```diff
@@ -112,25 +122,34 @@ def test_fill_forward_from_last_change_dataset(spark_session):
     expected_df = spark_session.createDataFrame(
         data=[
             # fmt: off
-                (1, 0, "2020-11-11",   "Yes",      1,      1,      1),
+                (1, 0, "2020-11-11",   "Yes",     1,         1,         1), # ensure that no filling occurs on participants with data from less than min datasets
                 (1, 0, "2020-11-12",   None,      None,      None,      None),
                 (1, 1, "2020-11-13",   None,      None,      None,      None),
 
-                (2, 1, "2020-11-11",   "Yes",      1,      1,      1),
-                (2, 1, "2020-11-12",   None,       1,      1,      1),
+                (2, 0, "2020-11-10",   "Yes",     1,         2,         1), # variable pattern will be picked up for fill forwards but not used
+                (2, 0, "2020-11-11",   None,      None,      None,      None), # no filling should occur on this row as dataset is below minimum
+                (2, 1, "2020-11-12",   "Yes",     1,         1,         1), # variable pattern will be picked up for fill forwards
+                (2, 1, "2020-11-13",   None,      1,         2,         1),
+                (2, 2, "2020-11-14",   None,      1,         1,         1), # not null values will be overwritten until next marked changed row
+                (2, 2, "2020-11-15",   None,      1,         1,         1), # correct values from row 6 should be seen in all 3 positions
+                (2, 3, "2020-11-16",   "Yes",     3,         3,         3), # variable changed within dataset should be carried forward
+                (2, 3, "2020-11-17",   None,      3,         3,         3),
+
+                (3, 0, "2020-11-11",   None,      1,         1,         1), # no marked rows so first should be filled forwards
+                (3, 3, "2020-11-18",   None,      1,         1,         1)
             # fmt: on
         ],
         schema="id integer, dataset integer, date string, changed string, var_1 integer, var_2 integer, var_3 integer",
     )
-    actual_df = fill_forward_from_last_change(
```

Comment:
Please could we also create a new test function using the original test data for the original version of the function, to make sure this still works as before